linked_list/implementation.py

```diff
@@ -7,34 +8,151 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
+        
+        self.elements = elements
+        
+        self.start = None
+        self.end = None
+        
+        if self.elements:
+            for element in self.elements:
+                self.append(element)
 
+                
     def __str__(self):
-        pass
+        return '[{}]'.format(", ".join(str(p) for p in self.elements)) if self.elements else "[]"
```

Comment:
good stuff. great way to manage the requirements here. excellent use of ternary operation.